docs/book/src/tasks/using-kustomize.md

```diff
@@ -0,0 +1,154 @@
+# Using Kustomize with Workload Cluster Manifests
+
+Although the `clusterctl config cluster` command exposes a number of different configuration values 
+for customizing workload cluster YAML manifests, some users may need additional flexibility above 
+and beyond what `clusterctl config cluster` or the example "flavor" templates that some CAPI providers 
+supply (as an example, see [these flavor templates](https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/master/templates/flavors) 
+for the Cluster API Provider for Azure). In the future, a [templating solution](https://github.com/kubernetes-sigs/cluster-api/issues/3252) 
+may be integrated into `clusterctl` to help address this need, but in the meantime users can use 
+`kustomize` as a solution to this need.
+
+This document provides a few examples of using `kustomize` with Cluster API. All of these examples 
+assume that the "base" (unmodified) Cluster API configuration (perhaps generated using 
+`clusterctl config cluster`) is in a directory whose relative path is `../../base`.
```

Comment:
Might be we should describe the the filesystem tree you are assuming, so `../../base` is a little bit more confusing.

Reply:
@fabriziopandini Have a look at the changes in 0f98ee3aa. I've added a directory tree/structure at the start of the document and clarified the wording.

Reply:
thanks!